core/src/main/scala/cromwell/core/logging/LoggingTest.scala

```diff
@@ -0,0 +1,7 @@
+package cromwell.core.logging
+
+import java.util.regex.Pattern
+
+object LoggingTest {
+  def escapePattern(pattern: String) = Pattern.quote(pattern)
```

Comment:
**TL;DR To use `LoggingTest` as a core test utility, add a dependency on `test->test` in build.sbt**

Yay more special maven/sbt syntax.

I'm assuming sbt/ivy is the same as maven, in that "[test … scope is not transitive](https://maven.apache.org/guides/introduction/introduction-to-dependency-mechanism.html#Dependency_Scope)". Only "compile [artifacts] are propagated to dependent projects".

In build.sbt, by default only main/`compile` code is added to child `compile` dependency trees. You need to explicitly tell the child artifact that it [depends on](http://www.scala-sbt.org/0.13/docs/Multi-Project.html#Per-configuration+classpath+dependencies) core's test code.

```scala
lazy val someArtifact = (project in file("someRoot"))
  .dependsOn(otherArtifact)
  .dependsOn(core % "test->test") // I don't need core's main code -or- I got core's main from otherArtifact transitively
```

or, if you also need the code from core's main folder in your child artifact's main code,

```scala
lazy val someArtifact = (project in file("someFolder"))
  .dependsOn(core % "compile->compile;test->test")  // "I want it all…" ♫
```

Looking through build.sbt, I believe what's happened is that we've correctly setup the dependency tree for our transitive main/`compile` code, but do _not_ have enough explicit setup for the non-transitive code living in test folders.